### PR TITLE
Make pin_memory of the Dataloader configurable

### PR DESCRIPTION
**Pull request by MeraX**
Pinned memory caused crashes on our machine, so we need this setting to be configurable.

### CODE REVIEW

**Comment by FussyDuck** [comment]
[CLA assistant check](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-training?pullRequest=64) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bol-claassistant.ecmwf.int:443/ecmwf/anemoi-training?pullRequest=64) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bol-claassistant.ecmwf.int:443/check/ecmwf/anemoi-training?pullRequest=64) it.</sub>

File: src/anemoi/training/data/datamodule.py [hunk 1]
```diff
@@ -96,6 +97,9 @@ def __init__(self, config: DictConfig) -> None:
             )
             self.config.dataloader.training.end = self.config.dataloader.validation.start - 1
 
+        if not self.config.dataloader.pin_memory:
```
Comment (HCookie) [inline src/anemoi/training/data/datamodule.py:100]: Please adjust this config retrieval to use a `.get` call with a default value to reduce config issues.
```python
if not self.config.dataloader.get('pin_memory', True):
```

File: src/anemoi/training/data/datamodule.py [hunk 2]
```diff
@@ -185,7 +189,7 @@ def _get_dataloader(self, ds: NativeGridDataset, stage: str) -> DataLoader:
             num_workers=self.config.dataloader.num_workers[stage],
             # use of pinned memory can speed up CPU-to-GPU data transfers
             # see https://pytorch.org/docs/stable/notes/cuda.html#cuda-memory-pinning
-            pin_memory=True,
+            pin_memory=self.config.dataloader.pin_memory,
```
Comment (HCookie) [inline src/anemoi/training/data/datamodule.py:192]: See above comment about using `.get`

**Review by HCookie — CHANGES_REQUESTED** [review]

File: src/anemoi/training/data/datamodule.py (hunk 1 above)
Comment (MeraX) [reply src/anemoi/training/data/datamodule.py]: Thanks for your quick review. I've implemented you change requests.

File: src/anemoi/training/data/datamodule.py (hunk 2 above)
Comment (MeraX) [reply src/anemoi/training/data/datamodule.py]: Thanks for your quick review. I've implemented you change requests.

**Review by HCookie — APPROVED** [review]

**Pull request merged**